src/layout/IHyprLayout.cpp

```diff
@@ -195,10 +195,13 @@ void IHyprLayout::onBeginDragWindow() {
 void IHyprLayout::onEndDragWindow() {
     const auto DRAGGINGWINDOW = g_pInputManager->currentlyDraggedWindow;
 
-    g_pInputManager->unsetCursorImage();
-
-    if (!g_pCompositor->windowValidMapped(DRAGGINGWINDOW))
+    if (!g_pCompositor->windowValidMapped(DRAGGINGWINDOW)) {
+        if (DRAGGINGWINDOW)
```

Comment:
this is always false

Reply:
getting rid of that line would mean if a window closes while being dragged the icon would be stuck in the dragging one. Although realistically nobody would ever encounter this situation, still yeet?

Reply:
huh you right. However, in that case, you should reset dragging window to a nullptr.

Reply:
done!